includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign.php

```diff
@@ -227,23 +281,32 @@ public function test( $post_id, $emails ) {
 				__( 'ActiveCampaign API credentials are missing.', 'newspack-newsletters' )
 			);
 		}
-		$sync_result = $this->sync( get_post( $post_id ) );
-		if ( is_wp_error( $sync_result ) ) {
-			return $sync_result;
+		$post = get_post( $post_id );
+		/** Create disposable campaign for sending a test. */
+		$campaign_name = sprintf( 'Test for %s', $this->get_campaign_name( $post ) );
+		$campaign      = $this->create_campaign( get_post( $post_id ), $campaign_name );
+		if ( is_wp_error( $campaign ) ) {
+			return $campaign;
 		}
-		$test_result = $this->api_request(
+		add_post_meta( $post_id, 'ac_test_campaign', $campaign['id'] );
+		$test_result = $this->api_v1_request(
 			'campaign_send',
 			'GET',
 			[
 				'query' => [
 					'type'       => 'html',
 					'action'     => 'test',
-					'campaignid' => $sync_result['campaign_id'],
+					'campaignid' => $campaign['id'],
 					'messageid'  => $sync_result['message_id'],
 					'email'      => implode( ',', $emails ),
 				],
 			]
 		);
+		/** Delete the disposable campaign. */
+		$delete_res = $this->delete_campaign( $campaing['id'], true );
+		if ( ! is_wp_error( $delete_res ) ) {
+			delete_post_meta( $post_id, 'ac_test_campaign', $campaign['id'] );
+		}
 		return $test_result;
```

Comment:
This `test` method returns a response in a different format compared to the other ESP `test` methods. Those other methods return a success message that is shown to the user in the editor UI:

<img width="1268" alt="Screen Shot 2022-06-23 at 11 33 17 AM" src="https://user-images.githubusercontent.com/2230142/175360482-704de3d3-9251-4be1-88f4-7b836443ccac.png">

Can we show a similar message upon successful test send for ActiveCampaign, too?

Reply:
Good call! Done in c1590e43361e2fe03f33efaef933761018cedd1c